src/openpgp.js

```diff
@@ -344,6 +344,65 @@ export function decryptSessionKey({ message, privateKey, password }) {
 }
 
 
+/////////////////////////////////////////////
+//                                         //
+//   Public key signing and verification   //
+//                                         //
+/////////////////////////////////////////////
+
+
+/**
+ * Signs a paublic key.
+ * @param  {Key} publicKey                 public key to be signed
+ * @param  {Key|Array<Key>} privateKeys    array of keys or single key with decrypted secret key data to sign public key
+ * @return {Promise<String|Key>}           Public key object in form:
+ *                                           { publicKey:Key, publicKeyArmored:String }
+ * @static
+ */
+export function signPublicKey({ publicKey, privateKeys }) {
+  checkKey(publicKey, 'publicKey');
+  privateKeys = toArray(privateKeys);
+
+  if (asyncProxy) { // use web worker if available
+    return asyncProxy.delegate('signPublicKey', { publicKey, privateKeys });
+  }
+
+  return execute(() => {
+
+    const signedPublicKey = publicKey.sign(privateKeys);
```

Comment:
Since the user needs to pass in the public Key object here anyway, they could just call `publicKey.sign(privateKeys)` directly.